core/src/main/scala/org/apache/spark/scheduler/dynalloc/ExecutorMonitor.scala

```diff
@@ -180,6 +181,15 @@ private[spark] class ExecutorMonitor(
     }
   }
 
+  def excludedExecutorCount: Int = {
+    if (conf.get(EXCLUDE_ON_FAILURE_ENABLED).getOrElse(false) &&
+        !conf.get(EXCLUDE_ON_FAILURE_KILL_ENABLED)) {
```

Comment:
Pull these boolean's out as instance variables instead of looking them up each time ?